ic10emu/build.rs

```diff
@@ -32,14 +18,14 @@ where
     pub deprecated: bool,
 }
 
-fn write_repr_enum<T: std::io::Write, I, P>(
+fn write_repr_enum<'a, T: std::io::Write, I, P>(
     writer: &mut BufWriter<T>,
     name: &str,
-    variants: &I,
+    variants: I,
     use_phf: bool,
 ) where
-    P: Display + FromStr,
-    for<'a> &'a I: IntoIterator<Item = (&'a String, &'a EnumVariant<P>)>,
+    P: Display + FromStr + 'a,
+    I: Iterator<Item = (&'a String, &'a EnumVariant<P>)>,
 {
```

Comment:
```suggestion
    I: IntoIterator<Item = &'a (String, EnumVariant<P>)>,
```

with this you wouldn't have to do `coll.iter().map(|(key, variant)| (key, variant))` and could just give `&coll`